bookwyrm/forms.py

```diff
@@ -226,6 +227,18 @@ class FileLinkForm(CustomForm):
     class Meta:
         model = models.FileLink
         fields = ["url", "filetype", "availability", "book", "added_by"]
+    
+    def clean(self):
+        """make sure the domain isn't blocked or pending"""
+        cleaned_data = super().clean()
+        url = cleaned_data.get('url')
+        domain = urlparse(url).netloc
+        if models.LinkDomain.objects.filter(domain=domain).exists():
+            status = models.LinkDomain.objects.get(domain=domain).status
+            if status == 'blocked':
+                self.add_error("url", _("Domain is blocked. Don't try this url again."))
+            elif status == 'pending':
+                self.add_error("url", _("Domain already pending. Please try later."))
```

Comment:
It's actually just fine if users add links where the domain is pending -- once the domain is reviewed, the links added while the domain was pending will be shown on the book page. So this check can be removed.